src/main/java/bot/telegram/state/StateProcessor.java

```diff
@@ -0,0 +1,109 @@
+package bot.telegram.state;
+
+import bot.domain.User;
+import bot.repository.UserService;
+import bot.telegram.keyboard.MainMenuKeyboard;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
+import org.telegram.telegrambots.meta.api.objects.Message;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+public class StateProcessor {
+    MessageSender sender;
+    MainMenuKeyboard menu;
+    UserService service;
+
+    public SendMessage processMessage(BotState state, Message message) {
+        String chatId = message.getChatId().toString();
+        switch (state) {
+
+            case WANNA_GET_SHARE:
+            case FIND_SHARE:
+                return processSearchShare(message);
+
+            case WANNA_GET_BOND:
+                return processSearchBond(message);
+
+            case MAKE_PORTFOLIO:
+            case GET_PORTFOLIO:
+            case DELETE_PORTFOLIO:
+            case UPDATE_PORTFOLIO:
+                return null;
+
+            case UNRECOGNIZED:
+            case GET_START_MENU:
+            case GET_HELP:
+                return processHintMessage(chatId);
+        }
+
+        return null;
+    }
+
+    private SendMessage processHintMessage(String chatId) {
+        User user = service.initializeUser(chatId);
+        BotState state = user.getState();
+        SendMessage reply = SendMessage.builder().chatId(chatId).text(state.name()).build();
+
+        switch (state) {
```

Comment:
@temalead Also consider to migrate from `switch` here, as I described above. You can utilize the polymorphism here and implement kind of `Strategy` pattern here.